gematria/granite/graph_builder_model_inference.cc

```diff
@@ -346,7 +420,54 @@ GraphBuilderModelInference::FromTfLiteModel(
   }
   llvm::Expected<std::unique_ptr<tflite::Interpreter>> interpreter =
       CreateInterpreter(*tflite_model);
-  if (auto error = interpreter.takeError()) return error;
+  if (llvm::Error error = interpreter.takeError()) return error;
+
+  // Get a mapping between the input tensors used in the model and their
+  // corresponding indices.
+  std::vector<int> input_tensor_to_idx(kNumPossibleInputTensors, -1);
+  for (int idx : (*interpreter)->inputs()) {
+    const std::string_view name = (*interpreter)->GetInputName(idx);
+    std::optional<InputTensor> input_tensor = GetInputTensorFromName(name);
+
+    // Report any unexpected tensors.
+    if (!input_tensor.has_value()) {
+      llvm::errs() << "Model contains unexpected tensor `" << name
+                   << "` at index " << idx << '\n';
+      continue;
+    }
+    input_tensor_to_idx.at(static_cast<int>(input_tensor.value())) = idx;
+  }
+
+  // Checks whether the model has all input tensors required to have for all
+  // graph builder models.
+  std::vector<InputTensor> missing_inputs;
+  for (const InputTensor input : kRequiredInputTensors) {
+    if (input_tensor_to_idx.at(static_cast<int>(input)) == -1)
+      missing_inputs.push_back(input);
+  }
```

Comment:
I'd add the check for the deltas tensor here, e.g.
```cc
if (uses_deltas && input_tensor_to_idx[InputTensor::kDeltaBlockIndexTensor] == -1) {
  missing_tensor.push_back(InputTensor::kDeltaBlockIndexTensor);
}
```

Reply:
I'm not sure that sounds right, since `uses_deltas` is defined as
```cpp
const bool uses_deltas = input_tensor_to_idx[static_cast<int>(
                              InputTensor::kDeltaBlockIndexTensor)] != -1;
```
i.e. it's the other way around, we infer what `uses_deltas` should be based on whether we found `kDeltaBlockIndexTensor` or not. I think [this response to a previous comment](https://github.com/google/gematria/pull/145#discussion_r1687962267) is relevant.

Reply:
Right, we can just skip this part.